### PR TITLE
Feature/7.6/code gen

### PR DESCRIPTION
**Pull request by Mpdreamz**
Downloads the spec and generates the `low level client`

Will follow up once merged with a PR for the `high level client`

### CODE REVIEW

File: src/ApiGenerator/RestSpecification/Core/delete_by_query.json [hunk 1]
```diff
@@ -178,9 +178,9 @@
         "description": "The throttle for this request in sub-requests per second. -1 means no throttle."
       },
       "slices": {
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/Core/delete_by_query.json]: BWC issue?

File: src/ApiGenerator/RestSpecification/XPack/ccr.forget_follower.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "ccr.forget_follower":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-forget-follower.html"
+      "url":"http://www.elastic.co/guide/en/elasticsearch/reference/current"
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/XPack/ccr.forget_follower.json]: Loss of information with url change?
https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-forget-follower.html seems valid still...

File: src/ApiGenerator/RestSpecification/XPack/ccr.unfollow.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "ccr.unfollow":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-unfollow.html"
+      "url":"http://www.elastic.co/guide/en/elasticsearch/reference/current"
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/XPack/ccr.unfollow.json]: Loss of information with url change?
https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-unfollow.html seems valid still...

File: src/ApiGenerator/RestSpecification/XPack/security.delete_privileges.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "security.delete_privileges":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-delete-privilege.html"
+      "url":"TODO"
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/XPack/security.delete_privileges.json]: Loss of information with url change?

File: src/ApiGenerator/RestSpecification/XPack/security.put_privileges.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "security.put_privileges":{
     "documentation":{
-      "url": "https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-privileges.html"
+      "url":"TODO"
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/XPack/security.put_privileges.json]: Loss of information with url change?

File: src/ApiGenerator/RestSpecification/XPack/sql.clear_cursor.json [hunk 6]
```diff
@@ -1,8 +1,7 @@
 {
   "sql.clear_cursor":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-pagination.html",
-      "description":"Clears the SQL cursor"
+      "url":"Clear SQL cursor"
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/XPack/sql.clear_cursor.json]: Bad merge on the ES side?

File: src/ApiGenerator/RestSpecification/XPack/sql.query.json [hunk 7]
```diff
@@ -1,8 +1,7 @@
 {
   "sql.query":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-rest-overview.html",
-      "description":"Executes an SQL request"
+      "url":"Execute SQL"
```
Comment (codebrain) [inline src/ApiGenerator/RestSpecification/XPack/sql.query.json]: Bad merge on the ES side?

File: src/ApiGenerator/RestSpecification/Core/delete_by_query.json (hunk 1 above)
Comment (Mpdreamz) [reply src/ApiGenerator/RestSpecification/Core/delete_by_query.json]: 517142a deals with this, we ignore the fact this is a union for now

File: src/ApiGenerator/RestSpecification/XPack/ccr.forget_follower.json (hunk 2 above)
Comment (Mpdreamz) [reply src/ApiGenerator/RestSpecification/XPack/ccr.forget_follower.json]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-forget-follower.html"
```

File: src/ApiGenerator/RestSpecification/XPack/ccr.unfollow.json (hunk 3 above)
Comment (Mpdreamz) [reply src/ApiGenerator/RestSpecification/XPack/ccr.unfollow.json]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-unfollow.html"
```

File: src/ApiGenerator/RestSpecification/XPack/security.delete_privileges.json (hunk 4 above)
Comment (Mpdreamz) [inline src/ApiGenerator/RestSpecification/XPack/security.delete_privileges.json]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-delete-privilege.html"
```

File: src/ApiGenerator/RestSpecification/XPack/security.put_privileges.json (hunk 5 above)
Comment (Mpdreamz) [reply src/ApiGenerator/RestSpecification/XPack/security.put_privileges.json]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-privileges.html"
```

File: src/ApiGenerator/RestSpecification/XPack/sql.clear_cursor.json (hunk 6 above)
Comment (Mpdreamz) [reply src/ApiGenerator/RestSpecification/XPack/sql.clear_cursor.json]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-rest-overview.html"
```

This is the way it is in `7.6.0` tag on ES

File: src/ApiGenerator/RestSpecification/XPack/sql.query.json (hunk 7 above)
Comment (Mpdreamz) [reply src/ApiGenerator/RestSpecification/XPack/sql.query.json]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-rest-overview.html"
```

The way it is on ES, will use the old url for now

**Comment by Mpdreamz** [comment]
Updated so that the links in code are always `https://` even if the docs indicate `http://`

**Pull request merged**